File: api-server/models/fingerprint.py
from pydantic import BaseModel, Field
from typing import List, Optional

class FingerprintAP(BaseModel):
    bssid: str
    rssi: int
    channel: Optional[int] = None

class FingerprintEntry(BaseModel):
    locationId: str
    timestamp: int
    scanId: int
    apList: List[FingerprintAP]

File: diagnostic-app/app/src/main/java/com/hyandlh/tarumtarenanavigation/core/model/Fingerprint.kt
package com.hyandlh.tarumtarenanavigation.core.model

/**
 * A recorded signal pattern at a specific location.
 */
data class FingerprintEntry(
    val locationId: String,
    val timestamp: Long,
    val scanId: Int,
    val apList: List<FingerprintAP>,
    val sessionId: String? = null,
    val orientation: String? = null
)

/**
 * A single AP reading within a fingerprint.
 */
data class FingerprintAP(
    val bssid: String,
    val rssi: Int,
    val channel: Int? = null
)
